client/post-editor/editor-gutenberg-opt-in-notice/index.jsx

```diff
@@ -30,20 +32,14 @@ class EditorGutenbergOptInNotice extends Component {
 		// connected properties
 		showDialog: PropTypes.func,
 		optInEnabled: PropTypes.bool,
+		noticeDismissed: PropTypes.bool,
+		dismissNotice: PropTypes.func,
 	};
 
-	state = {
-		dismissed: false,
-	};
-
-	dismissNotice = () => this.setState( { dismissed: true } );
+	dismissNotice = () => this.props.dismissNotice( 'gutenberg_nudge_notice_dismissed', true );
```

Comment:
One thing I'd consider is saving the timestamp of dismissal here instead of just true. Just in case someone decides in a month that we should boost opt-in rates by showing this nudge again every month or so.
It would make things easier to handle, instead of grandfathering. Your call. :)

Reply:
I'll stick with what we have, and we can deal with it if it ever comes to pass 🤷‍♂️